numpy/core/src/multiarray/getset.c

```diff
@@ -549,7 +568,7 @@ array_descr_set(PyArrayObject *self, PyObject *arg, void *NPY_UNUSED(ignored))
             PyArray_DIMS(self)[axis] *= newdim;
             PyArray_STRIDES(self)[axis] = newtype->elsize;
         }
-        else if (newtype->elsize > PyArray_DESCR(self)->elsize) {
```

Comment:
This was redundant because the whole conditional is nested in the check `if (newtype->elsize != PyArray_DESCR(self)->elsize) {` on line 498.

Reply:
Maybe change it to an `assert` just to make it easy to see that.